modules/audit-enqueued-assets/load.php

```diff
@@ -0,0 +1,312 @@
+ *
+ * @return array
+ */
+function perflab_aea_enqueued_css_assets_test() {
+	/**
+	 * If the test didn't run yet, deactivate.
+	 */
+	$enqueued_styles = perflab_aea_get_total_enqueued_styles();
+	if ( false === $enqueued_styles ) {
+		return array();
+	}
+	$result = array(
+		'label'       => esc_html__( 'Enqueued CSS assets', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => esc_html__( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__( 'The amount of enqueued CSS assets is acceptable.', 'performance-lab' )
+		),
+		'actions'     => '',
+		'test'        => 'enqueued_css_assets',
+	);
+
+	if ( $enqueued_styles > 10 ) {
+		$result['status']         = 'recommended';
+		$result['badge']['color'] = 'orange';
+		$result['description']    = sprintf(
+			/* translators: %s: Number of enqueued styles */
+			esc_html__( 'Your website enqueues %s styles. Try to reduce the number of CSS assets, or to concatenate them.', 'performance-lab' ),
+			$enqueued_styles
+		);
+
+		$result['actions'] .= sprintf(
+			/* translators: 1: HelpHub URL. 2: Link description. 3.URL to clean cache. 4. Clean Cache text. */
+			'<p><a target="_blank" href="%1$s">%2$s</a></p><p><a href="%3$s">%4$s</a></p>',
+			esc_url( __( 'https://wordpress.org/support/article/optimization/', 'performance-lab' ) ),
+			esc_html__( 'More info about performance optimization', 'performance-lab' ),
+			esc_url( add_query_arg( 'action', 'clean_aea_audit', wp_nonce_url( admin_url( 'site-health.php' ), 'clean_aea_audit' ) ) ),
+			esc_html__( 'Clean Test Cache', 'performance-lab' )
+		);
+	}
+
+	return $result;
+}
+
+/**
+ * Invalidate both transients/cache on user clean_aea_audit action.
+ * Redirects to site-health.php screen adter clean up.
+ *
+ * @since 1.0.0
+ */
+function perflab_clean_aea_audit_action() {
+	if ( isset( $_GET['action'] ) && 'clean_aea_audit' === $_GET['action'] ) {
+		check_admin_referer( 'clean_aea_audit' );
+		perflab_invalidate_cache_transients();
+		wp_safe_redirect( remove_query_arg( array( 'action', '_wpnonce' ), wp_get_referer() ) );
+	}
+}
+add_action( 'admin_init', 'perflab_clean_aea_audit_action' );
+
+/**
+ * Invalidate both transients/cache.
+ *
+ * @since 1.0.0
+ */
+function perflab_invalidate_cache_transients() {
+	delete_transient( 'aea_enqueued_scripts' );
+	delete_transient( 'aea_enqueued_styles' );
+}
+add_action( 'switch_theme', 'perflab_invalidate_cache_transients' );
+add_action( 'activated_plugin', 'perflab_invalidate_cache_transients' );
+add_action( 'deactivated_plugin', 'perflab_invalidate_cache_transients' );
+
+
+
+/**
+ * Convert full URL paths to absolute paths.
+ *
+ * @since 1.0.0
+ *
+ * @param string $resource_url URl resource link.
+ * @return string Returns abosulte path to the resource.
+ */
+function perflab_get_path_from_resource_url( $resource_url ) {
+	return ABSPATH . wp_make_link_relative( $resource_url );
```

Comment:
Note: this needs to be tested with various WP setup, url to path is a tricky thing to nail.